src/wagtail_vector_index/index/models.py

```diff
@@ -0,0 +1,409 @@
+        }
+        return list(embedding_fields.values())
+
+    @classmethod
+    def check(cls, **kwargs):
+        """Extend model checks to include validation of embedding_fields in the
+        same way that Wagtail's Indexed class does it."""
+        errors = super().check(**kwargs)
+        errors.extend(cls._check_embedding_fields(**kwargs))
+        return errors
+
+    @classmethod
+    def _has_field(cls, name):
+        try:
+            cls._meta.get_field(name)
+        except FieldDoesNotExist:
+            return hasattr(cls, name)
+        else:
+            return True
+
+    @classmethod
+    def _check_embedding_fields(cls, **kwargs):
+        errors = []
+        for field in cls._get_embedding_fields():
+            message = "{model}.embedding_fields contains non-existent field '{name}'"
+            if not cls._has_field(field.field_name):
+                errors.append(
+                    checks.Warning(
+                        message.format(model=cls.__name__, name=field.field_name),
+                        obj=cls,
+                        id="wagtailai.WA001",
+                    )
+                )
+        return errors
+
+
+IndexedType = TypeVar("IndexedType")
+
+
+class EmbeddableFieldsDocumentConverter:
+    """Implementation of DocumentConverter that knows how to convert a model instance using the
+    EmbeddableFieldsMixin to and from a Document.
+
+    Stores and retrieves embeddings from an Embedding model."""
+
+    def __init__(self, base_model: type[models.Model]):
+        # The model that this converter will convert Documents back to
+        self.base_model = base_model
+
+    def _get_split_content(
+        self, object: EmbeddableFieldsMixin, *, chunk_size: int
+    ) -> list[str]:
+        """Split the contents of a model instance's `embedding_fields` in to smaller chunks"""
+        splittable_content = []
+        important_content = []
+        embedding_fields = object._meta.model._get_embedding_fields()
+
+        for field in embedding_fields:
+            value = field.get_value(object)
+            if value is None:
+                continue
+            if isinstance(value, str):
+                final_value = value
+            else:
+                final_value: str = "\n".join((str(v) for v in value))
+            if field.important:
+                important_content.append(final_value)
+            else:
+                splittable_content.append(final_value)
+
+        text = "\n".join(splittable_content)
+        important_text = "\n".join(important_content)
+        splitter = self._get_text_splitter(chunk_size=chunk_size)
+        return [f"{important_text}\n{text}" for text in splitter.split_text(text)]
+
+    def _get_text_splitter(
+        self, chunk_size: int
+    ) -> LangchainRecursiveCharacterTextSplitter:
+        length_calculator = NaiveTextSplitterCalculator()
+        return LangchainRecursiveCharacterTextSplitter(
```

Comment:
**Suggestion:** Make the text splitter class configurable.

Reply:
We could specify the text splitter class as the converter class property, rather than the user having to override a private method. I think that might be a nicer experience.

I can kind of see now that we probably don't want to create a base class with that logic for now so it's probably not worth doing.

Reply:
Yeah it's probably a nicer experience but I think something we can revisit.